pkg/credentialprovider/azure_credentials.go

```diff
@@ -163,13 +197,21 @@ func (a *acrProvider) GetCredentials(_ context.Context, image string, _ []string
 }
 
 // getFromACR gets credentials from ACR.
-func (a *acrProvider) getFromACR(loginServer string) (string, string, error) {
+func (a *acrProvider) getFromACR(ctx context.Context, loginServer string) (string, string, error) {
+	config, err := azclient.GetAzureCloudConfig(&a.config.ARMClientConfig)
+	if err != nil {
+		return "", "", err
+	}
+	var armAccessToken azcore.AccessToken
 	// Run EnsureFresh to make sure the token is valid and does not expire
-	if err := a.servicePrincipalToken.EnsureFresh(); err != nil {
+	if armAccessToken, err = a.credential.GetToken(ctx, policy.TokenRequestOptions{
+		Scopes: []string{
+			config.Services[azcontainerregistry.ServiceName].Audience,
```

Comment:
Does the current azure.json or somewhere else provide this Audience config?

Reply:
yes. It is defined in sdk and overwritten by value in environment config.

Reply:
Please correct me if I am wrong, in AKS case, it's defined in SDK, right? because we don't have these values explicitly defined in environment config. Please provide the link of the SDK or post the environment config example to help me understand the user case.
Thanks.

Reply:
All of endpoint/audience config is defined in [cloud package](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.13.0/cloud). And each package will add service config in [init function](https://github.com/Azure/azure-sdk-for-go/blob/sdk/containers/azcontainerregistry/v0.2.1/sdk/containers/azcontainerregistry/cloud_config.go). 

But config will be overwritten if environment file is provided. 
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/c6a16f0f757ec52aef0c816ff75dbcb6af8c1b1b/pkg/azclient/cloud.go#L120